src/basic/initrd-util.c

```diff
@@ -33,8 +27,7 @@ bool in_initrd(void) {
         if (r >= 0)
                 saved_in_initrd = r > 0;
         else
-                saved_in_initrd = access("/etc/initrd-release", F_OK) >= 0 &&
-                                  path_is_temporary_fs("/") > 0;
+                saved_in_initrd = access("/etc/initrd-release", F_OK) >= 0;
```

Comment:
(in newer code we usually add debug logging to unexpected errors, hence, if you want to win extra points you'd change this to:

```c
r = RET_NERRNO(access("/etc/initrd-release", F_OK));
if (r < 0 && r != -ENOENT)
        log_debug_errno(r, "Failed to check if /etc/initrd-release exists, assuming it does not: %m");

save_in_initrd = r >= 0;
```

or something like that